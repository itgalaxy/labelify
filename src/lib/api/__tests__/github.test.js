"use strict";

const githubAPI = require("../github");
const nock = require("nock");

const { createLabel, getLabels, removeLabel, updateLabel } = githubAPI;

describe("github api", () => {
  const url = "https://api.github.com";
  const projectID = "itgalaxy/labelify";
  const endpointPart = `/repos/${projectID}/labels`;
  const endpoint = `${url}${endpointPart}`;
  const config = { endpoint, token: "token" };

  it("should success create the label", () => {
    expect.assertions(1);

    nock(url)
      .post(endpointPart)
      .reply(201, {
        color: "fff",
        default: false,
        description: "description",
        id: 1,
        name: "name",
        url: `${endpoint}/name`
      });

    return expect(
      createLabel(config, {
        color: "#fff",
        description: "description",
        name: "name"
      })
    ).resolves.toMatchSnapshot();
  });

  it("should success create the label without description", () => {
    expect.assertions(1);

    nock(url)
      .post(endpointPart)
      .reply(201, {
        color: "fff",
        default: false,
        description: null,
        id: 1,
        name: "name",
        url: `${endpoint}/name`
      });

    return expect(
      createLabel(config, {
        color: "#fff",
        name: "name"
      })
    ).resolves.toMatchSnapshot();
  });

  it("should success get labels", () => {
    expect.assertions(1);

    nock(url)
      .get(endpointPart)
      .reply(200, [
        {
          color: "fff",
          default: false,
          description: "description",
          id: 1,
          name: "name",
          url: `${endpoint}/name`
        },
        {
          color: "fff",
          default: false,
          description: "other description",
          id: 2,
          name: "other-name",
          url: `${endpoint}/other-name`
        }
      ]);

    return expect(getLabels(config)).resolves.toMatchSnapshot();
  });

  it("should success remove the label", () => {
    expect.assertions(1);

    nock(url)
      .delete(`${endpointPart}/name`)
      .reply(204, "");

    return expect(
      removeLabel(config, {
        name: "name"
      })
    ).resolves.toMatchSnapshot();
  });

  it("should success update the label", () => {
    expect.assertions(1);

    nock(url)
      .patch(`${endpointPart}/name`)
      .reply(200, {
        color: "000",
        default: false,
        description: "other-description",
        id: 1,
        name: "name",
        url: `${endpoint}/name`
      });

    return expect(
      updateLabel(config, {
        color: "fff",
        description: "description",
        name: "name"
      })
    ).resolves.toMatchSnapshot();
  });

  it("should success update the label without description", () => {
    expect.assertions(1);

    nock(url)
      .patch(`${endpointPart}/name`)
      .reply(200, {
        color: "000",
        default: false,
        description: null,
        id: 1,
        name: "name",
        url: `${endpoint}/name`
      });

    return expect(
      updateLabel(config, {
        color: "#fff",
        name: "name"
      })
    ).resolves.toMatchSnapshot();
  });
});
