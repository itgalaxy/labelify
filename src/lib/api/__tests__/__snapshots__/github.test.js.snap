// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github api should success create the label 1`] = `
Object {
  "color": "fff",
  "default": false,
  "description": "description",
  "id": 1,
  "name": "name",
  "url": "https://api.github.com/repos/itgalaxy/labelify/labels/name",
}
`;

exports[`github api should success create the label without description 1`] = `
Object {
  "color": "fff",
  "default": false,
  "description": null,
  "id": 1,
  "name": "name",
  "url": "https://api.github.com/repos/itgalaxy/labelify/labels/name",
}
`;

exports[`github api should success get labels 1`] = `
Array [
  Object {
    "color": "fff",
    "default": false,
    "description": "description",
    "id": 1,
    "name": "name",
    "url": "https://api.github.com/repos/itgalaxy/labelify/labels/name",
  },
  Object {
    "color": "fff",
    "default": false,
    "description": "other description",
    "id": 2,
    "name": "other-name",
    "url": "https://api.github.com/repos/itgalaxy/labelify/labels/other-name",
  },
]
`;

exports[`github api should success remove the label 1`] = `""`;

exports[`github api should success update the label 1`] = `
Object {
  "color": "000",
  "default": false,
  "description": "other-description",
  "id": 1,
  "name": "name",
  "url": "https://api.github.com/repos/itgalaxy/labelify/labels/name",
}
`;

exports[`github api should success update the label without description 1`] = `
Object {
  "color": "000",
  "default": false,
  "description": null,
  "id": 1,
  "name": "name",
  "url": "https://api.github.com/repos/itgalaxy/labelify/labels/name",
}
`;
